packages/icons/src/icon/index.js

```diff
@@ -3,6 +3,20 @@
  */
 import { cloneElement } from '@wordpress/element';
 
+// Disable reason: JSDoc linter doesn't seem to parse the union (`&`) correctly.
+/* eslint-disable jsdoc/valid-types */
+/** @typedef {{icon: JSX.Element, size?: number} & import('react').ComponentPropsWithoutRef<'SVG'>} IconProps */
```

Comment:
This isn't _strictly_ accurate, but this is closer to what we intend. I believe `ComponentPropsWithoutRef<'SVG'>` is a bug.

```js
/** @typedef {{icon: JSX.Element, size?: number} & import('@wordpress/primitives').SVGProps} IconProps */
```